Configurations.md

```diff
@@ -2294,6 +2294,39 @@ use std::sync::Arc;
 use uuid::Uuid;
 ```
 
+#### Example for custom groups:
+
+Discard existing import groups, and create groups as specified by the wildcarded list.
+Handy aliases are supported:
+
+- `$std` prefix is an alias for standard library (i.e `std`, `core`, `alloc`);
+- `$crate` prefix is an alias for crate-local modules (i.e `self`, `crate`, `super`).
+
+For a given config:
+
+```toml
+group_imports = [
```

Comment:
It might be handy to document how one could create a group for any crate not already in a group, like in `StdExternalCrate`. Based on the tests for this, I think this might suffice:
```toml
group_imports = [
    ["$std::*", "proc_macro::*"],
    # external crates
    ["*"],
    ["my_crate::*", "crate::*::xyz"],
    ["$crate::*"],
]
```

Reply:
Current implementation won't work for your example as you might expect. The groups are checked for a match consequently thus all that doesn't match the first group goes to the second with the last two being always unused.

I don't really think changing implementation for supporting this single case for external crates is a good idea, nor handling myriad of edge-cases for a general solution. But perhaps a note about order would be handy.

Reply:
I've raised a PR in your repo for adding custom fallback groups: https://github.com/l4l/rustfmt/pull/1. I think the approach is simple enough that it can be included with this feature. Without this, we're limited to StdWorkspaceCrateExternal (we can add regexs for all external crates, but that's not practical). I don't see myself using this feature (group_imports) if I can't do StdExternalWorkspaceCrate.

Reply:
Agreed with @PatchMixolydic and @utkarshgupta137, being able to specify where "everything else" goes in a simple way without having to jump through hoops seems incredibly useful.

Reply:
Added @utkarshgupta137 patch with an addition of that group being restricted to a single occurrence. Thanks for the opinions and the PR